src/stream.ts

```diff
@@ -3143,88 +3169,67 @@ export class Part extends Stream {
+            [systemCurrentWidths, systemBreakIndexes] = this.systemWidthsAndBreaks();
+        }
+        else {
+            // read from measure render options
+            let lastSystemIndex = 0;
+            let workingSystemWidth = 0;
+            for (const [i, m] of Array.from(this.measures).entries()) {
+                if (m.renderOptions.systemIndex === lastSystemIndex) {
+                    workingSystemWidth += m.renderOptions.width;
+                } else {
+                    systemCurrentWidths.push(workingSystemWidth);
+                    systemBreakIndexes.push(i - 1);
+                    workingSystemWidth = m.renderOptions.width;
+                }
+                lastSystemIndex = m.renderOptions.systemIndex;
+            }
+        }
         if (systemPadding === undefined) {
             systemPadding = this.renderOptions.systemPadding;
         }
-        const maxSystemWidth = this.maxSystemWidth; // cryptic note: 'of course fix!'?
+        // last system is not contained in systemCurrentWidths
+        // possible, but unlikely that the last system would be the
+        // longest and thus left out of the calculation
+        const maxSystemWidth = Math.max(...systemCurrentWidths);
 
         let currentSystemIndex = 0;
-        let leftSubtract = 0;
-        let newLeftSubtract;
-
-        for (const [i, el] of Array.from(this.getElementsByClass('Measure')).entries()) {
-            const m = el as Measure;
-            if (i === 0) {
-                m.renderOptions.startNewSystem = true;
-            }
-            const currentLeft = m.renderOptions.left;
-
-            if (systemBreakIndexes.indexOf(i - 1) !== -1) {
-                /* first measure of new System */
-                const oldWidth = m.renderOptions.width;
-                const oldEstimate = m.estimateStaffLength() + m.renderOptions.staffPadding;
-                const offsetFromEstimate = oldWidth - oldEstimate;
-                // we look at the offset from the current estimate to see how much
-                // the staff length may have been adjusted to compensate for other
-                // parts with different lengths.
-
-                // but setting these options is bound to change something
-                m.renderOptions.displayClef = true;
-                m.renderOptions.displayKeySignature = true;
-                m.renderOptions.startNewSystem = true;
-
-                // so we get a new estimate.
-                const newEstimate = m.estimateStaffLength() + m.renderOptions.staffPadding;
-
-                // and adjust it for the change.
-                const newWidth = newEstimate + offsetFromEstimate;
-                m.renderOptions.width = newWidth;
-                leftSubtract = currentLeft - 20;
-                // after this one, we'll have a new left subtract...
-                newLeftSubtract = leftSubtract - (newWidth - oldWidth);
 
+        for (const [i, m] of Array.from(this.measures).entries()) {
```

Comment:
I'm trying to get away from using `this.measures` internally, because eventually .measures should be equivalent to the music21p method.